components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/OrganizationManagementDAOImpl.java

```diff
@@ -709,97 +726,82 @@ public String resolveTenantDomain(String organizationId) throws OrganizationMana
-                        ? StringUtils.isNotBlank(applicationAudience)
-                        ? GET_ORGANIZATIONS_WITH_USER_ROLE_ASSOCIATIONS_TAIL_MSSQL
-                        : GET_ORGANIZATIONS_WITH_USER_ASSOCIATIONS_TAIL_MSSQL
-                        : GET_ORGANIZATIONS_TAIL_MSSQL;
-            }
+        String userID =  getUserId();
+        String sqlStmt = prepareGetOrganizationQuery(authorizedSubOrgsOnly, recursive, sortOrder,
+                applicationAudience, filterQueryBuilder, parentIdFilterQueryBuilder, userID);
 
+        List<String> permissions;
+        String permissionPlaceholder;
+        if (CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME) {
+            permissionPlaceholder = "PERMISSION_";
+            permissions = getAllowedPermissions(VIEW_ORGANIZATION_PERMISSION);
             if (authorizedSubOrgsOnly) {
-                if (StringUtils.isNotBlank(applicationAudience)) {
-                    sqlStmt = GET_ORGANIZATIONS_WITH_USER_ROLE_ASSOCIATIONS;
-                } else {
-                    sqlStmt = GET_ORGANIZATIONS_WITH_USER_ASSOCIATIONS;
-                }
-            } else {
-                sqlStmt = GET_ORGANIZATIONS;
+                sqlStmt = replacePermissionPlaceholders(sqlStmt, permissions, permissionPlaceholder);
             }
         } else {
-            getOrgSqlStmtTail = authorizedSubOrgsOnly ? GET_ORGANIZATIONS_TAIL_LEGACY
-                    : GET_ORGANIZATIONS_TAIL_WITHOUT_PERMISSION_CHECK;
-
-            if (isOracleDB()) {
-                getOrgSqlStmtTail = authorizedSubOrgsOnly ? GET_ORGANIZATIONS_TAIL_ORACLE_LEGACY
-                        : GET_ORGANIZATIONS_TAIL_ORACLE_WITHOUT_PERMISSION_CHECK;
-            } else if (isMSSqlDB()) {
-                getOrgSqlStmtTail = authorizedSubOrgsOnly ? GET_ORGANIZATIONS_TAIL_MSSQL_LEGACY
-                        : GET_ORGANIZATIONS_TAIL_MSSQL_WITHOUT_PERMISSION_CHECK;
-            }
+            permissionPlaceholder = "";
+            permissions = new ArrayList<>();
+        }
 
-            if (authorizedSubOrgsOnly) {
-                sqlStmt = GET_ORGANIZATIONS_LEGACY;
-            } else {
-                sqlStmt = GET_ORGANIZATIONS_WITHOUT_PERMISSION_CHECK;
-            }
+        List<Organization> organizations;
+        NamedJdbcTemplate namedJdbcTemplate = Utils.getNewTemplate();
+        try {
+            organizations = namedJdbcTemplate.executeQuery(sqlStmt,
+                    (resultSet, rowNumber) -> {
+                        Organization organization = new Organization();
+                        organization.setId(resultSet.getString(1));
+                        organization.setName(resultSet.getString(2));
+                        organization.setCreated(resultSet.getTimestamp(3).toInstant());
+                        organization.setStatus(resultSet.getString(4));
+                        int columnCount = resultSet.getMetaData().getColumnCount();
+                        if (columnCount >= 6) {
```

Comment:
Find a solution to return all of the attributes belonging to that organization.

Reply:
This is now updated to return all of the attributes belonging to that organization.